### ISSUE TITLE
support removal of desktop shortcut from sidebar

### ISSUE THREAD

**Issue by brainchild0** [linuxmint/nemo#2407]
# Versions 

 * **Nemo version:** 4.4.2
 * **Desktop or windowed nemo?** Windowed
 * **Distribution:** Mint 19.3
 * **Architecture:** 64-bit

# Issue

No support is available for removing the home or desktop directories from the sidebar locations. Minimally, it would be helpful to  support  removing the desktop shortcut, since many users keep no content in this location, or otherwise have no need for convenient access.

**Comment by mtwebster** [comment]
try this:
```
gsettings set org.nemo.preferences desktop-is-home-dir true
```

**Comment by brainchild0** [comment]
@mtwebster The configuration has the effect requested, but also the unwanted effect of showing the contents of the home directory on the actual desktop of the workspace.

**Comment by mtwebster** [comment]
You can turn off the desktop icons - go to Desktop settings, set the desktop layout to 'no icons'

**Comment by brainchild0** [comment]
@mtwebster Yes, this combination is helpful. I would consider it a workaround. Requested is a  setting or command that  a user may  find easily in the Nemo interface. Giving the desktop folder the same handling as the other folders in the sidebar, with a *remove* option in the popup menu, seems that it would be a clear and compelling resolution.

**Comment by andrewdbate** [comment]
I was just about to open a feature request to ask for the same but then I found this issue.

In my case I had already turned off icons on the desktop with:
```
$ gsettings set org.nemo.desktop desktop-layout false::false
```
I wanted to remove the Desktop folder from the places sidebar because that folder is unused in this setup (because I don't keep anything on the desktop).

The workaround in this case (as commented above already) is to additionally set:
```
$ gsettings set org.nemo.preferences desktop-is-home-dir true
```

This sets the users home directory as the desktop (which was already set not to display anyway).

Screenshot **before** setting `desktop-is-home-dir true`:
![Before](https://user-images.githubusercontent.com/2998569/145262637-f0555542-0f4c-479e-a051-6b95108996ca.png)

Screenshot **after** setting `desktop-is-home-dir true`:
![After](https://user-images.githubusercontent.com/2998569/145262616-d30171e0-79dc-4e3e-b852-7f59ea046601.png)

However I agree with @brainchild0 that this is a bit of a hack / workaround. This workaround happens to fit my use case, but ideally you should be able to remove the Desktop folder from the places sidebar with a single `gsettings` key.

Note also that you can currently remove any other item from the places sidebar by editing the list in `Bookmarks` > `Edit Bookmarks...`. The only item you can't currently remove is the `Desktop` item which seems inconsistent because you can still remove the Home folder from this list after setting `desktop-is-home-dir true`.

**Comment by u2n** [comment]
3.7 years after this request was filed, nemo has progressed from 4.4.2 to 6.0.2, many improvements have been made (thank you) -- but not this one!

Surprising to find this since giving users the ability to customize the UI as they like is a hallmark of open source (as opposed to the iron-fisted MS/Apple approach). Commercial entities are expected to dictate _how_ their apps are used. That's them, not us.

Besides the Home and Desktop shortcuts at top, Recent, File System and Trash at bottom are also fixed. Hope this initiative gets some attention soon.
![Untitled](https://github.com/linuxmint/nemo/assets/5048383/da2ddcc0-1b1a-4676-af55-03fab900b253)

**Comment by 0587934204938** [comment]
please fix soon

**Comment by Jeremy7701** [comment]
Fixed in nemo 6.6.3 (or earlier) please close.
